dali/core/mm/malloc_resource.cc

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mutex>
+#include <unordered_map>
+#include "dali/core/cuda_stream_pool.h"
+#include "dali/core/device_guard.h"
+#include "dali/core/spinlock.h"
+#include "dali/core/mm/malloc_resource.h"
+#include "dali/core/mm/detail/align.h"
+
+namespace dali {
+namespace mm {
+
+#if CUDA_VERSION >= 11020
+
+namespace {
+
+class aligned_alloc_helper {
+ public:
+  static constexpr size_t kCudaMallocAlignment = 256;
```

Comment:
Is it given once for all or may change depending on the version?

Reply:
I think it's given - I asked about it on CUDA channel and all functions that allocate global memory return data aligned to at least 256B.